Modules/ITS/include/ITS/ITSClusterTask.h

```diff
@@ -51,75 +51,100 @@ class ITSClusterTask : public TaskInterface
   void endOfActivity(Activity& activity) override;
   void reset() override;
 
+  // coarse checks
+  void setRphiBinningIB(std::vector<float> bins = { -0.75, -0.60, -0.45, -0.30, -0.15, 0, 0.15, 0.30, 0.45, 0.60, 0.76 });
+  void setZBinningIB(std::vector<float> bins = { -1.5, -1.20, -0.9, -0.6, -0.3, 0, 0.3, 0.6, 0.9, 1.2, 1.51 });
+  void setRphiBinningOB(std::vector<float> bins = { -0.75, -0.35, 0, 0.35, 0.76 });
+  void setZBinningOB(std::vector<float> bins = { -1.5, -0.75, 0., 0.75, 1.51 });
+
  private:
   void publishHistos();
   void formatAxes(TH1* h, const char* xTitle, const char* yTitle, float xOffset = 1., float yOffset = 1.);
   void addObject(TObject* aObject);
   void getJsonParameters();
   void createAllHistos();
-  void updateOccMonitorPlots();
+
+  float getHorizontalBin(float z, int chip, int layer, int lane = 0);
+  float getVerticalBin(float rphi, int stave, int layer);
 
   static constexpr int NLayer = 7;
   static constexpr int NLayerIB = 3;
-  TH2D* hClusterVsBunchCrossing;
+
   std::vector<TObject*> mPublishedObjects;
-  TH1D* hClusterSizeSummaryIB[7][48][9];
-  TH1D* hClusterTopologySummaryIB[7][48][9];
-  TH1D* hGroupedClusterSizeSummaryIB[7][48][9];
 
-  TH1D* hClusterSizeLayerSummary[7];
-  TH1D* hClusterTopologyLayerSummary[7];
-  TH1D* hGroupedClusterSizeLayerSummary[7];
+  // Inner barrel
+  TH1F* hClusterTopologySummaryIB[NLayer][48][9] = { { { nullptr } } };
+  TH1F* hGroupedClusterSizeSummaryIB[NLayer][48][9] = { { { nullptr } } };
+
+  TH2F* hAverageClusterOccupancySummaryIB[NLayer] = { nullptr };
+  TH2F* hAverageClusterSizeSummaryIB[NLayer] = { nullptr };
+
+  int mClusterOccupancyIB[NLayer][48][9] = { { { 0 } } };
+
+  // Outer barrel
+  TH1F* hGroupedClusterSizeSummaryOB[NLayer][48] = { { nullptr } };
+  TH1F* hClusterSizeSummaryOB[NLayer][48] = { { nullptr } };
+  TH1F* hClusterTopologySummaryOB[NLayer][48] = { { nullptr } };
+
+  TH2F* hAverageClusterOccupancySummaryOB[NLayer] = { nullptr };
+  TH2F* hAverageClusterSizeSummaryOB[NLayer] = { nullptr };
+
+  int mClusterOccupancyOB[NLayer][48][28] = { { { 0 } } };
+
+  // Layer synnary
+  TH1F* hClusterSizeLayerSummary[NLayer] = { nullptr };
+  TH1F* hClusterTopologyLayerSummary[NLayer] = { nullptr };
+  TH1F* hGroupedClusterSizeLayerSummary[NLayer] = { nullptr };
+
+  // General
+  TH2F* hClusterVsBunchCrossing = nullptr;
+  TH2F* mGeneralOccupancy = nullptr;
 
-  TH2D* hAverageClusterOccupancySummaryIB[7];
-  TH2D* hAverageClusterOccupancyMonitorIB[7]; // will be used in online data monitoring, showing occupation for the last N ROFs
-  TH2D* hAverageClusterSizeSummaryIB[7];
-  TH2D* hAverageClusterSizeMonitorIB[7];
+  // Coarse checks
+  TH2F* hAverageClusterOccupancySummaryCoarse[NLayer] = { nullptr };
+  TH2F* hAverageClusterSizeSummaryCoarse[NLayer] = { nullptr };
```

Comment:
a curiosity, why `coarse` if they have actually a finer binning wrt the ones we have already in the task?